plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTableBuilder.java

```diff
@@ -93,33 +105,48 @@ public static DynamicTable buildFromPql(PinotMetadata pinotMetadata, SchemaTable
-            for (AggregationInfo aggregationInfo : request.getAggregationsInfo()) {
-                String baseColumnName = aggregationInfo.getAggregationParams().get(COLUMN_KEY);
-                AggregationExpression aggregationExpression;
-                if (baseColumnName.equals(WILDCARD)) {
-                    aggregationExpression = new AggregationExpression(getOutputColumnName(aggregationInfo, baseColumnName),
-                            baseColumnName,
-                            aggregationInfo.getAggregationType());
-                }
-                else {
-                    PinotColumnHandle columnHandle = (PinotColumnHandle) columnHandles.get(baseColumnName);
-                    if (columnHandle == null) {
-                        throw new ColumnNotFoundException(schemaTableName, aggregationInfo.getAggregationParams().get(COLUMN_KEY));
-                    }
-                    aggregationExpression = new AggregationExpression(
-                            getOutputColumnName(aggregationInfo, columnHandle.getColumnName()),
-                            columnHandle.getColumnName(),
-                            aggregationInfo.getAggregationType());
-                }
-
-                aggregationExpressionBuilder.add(aggregationExpression);
+            for (AggregationFunction aggregationFunction : queryContext.getAggregationFunctions()) {
+                aggregationFunction.getResultColumnName();
+                aggregationFunction.getType().getName();
+                aggregateColumnsBuilder.add(new PinotColumnHandle(
+                        aggregationFunction.getResultColumnName(),
+                        toTrinoType(aggregationFunction.getFinalResultColumnType())));
             }
         }
 
-        return new DynamicTable(pinotTableName, suffix, selectionColumns, groupByColumns, filter, aggregationExpressionBuilder.build(), orderBy, getTopNOrLimit(request), getOffset(request), query);
+        return new DynamicTable(pinotTableName, suffix, selectionColumns, filter, groupByColumns, aggregateColumnsBuilder.build(), orderBy, getTopNOrLimit(request), getOffset(request), query);
+    }
+
+    private static Type toTrinoType(DataSchema.ColumnDataType columnDataType)
+    {
+        switch (columnDataType) {
+            case INT:
+                return INTEGER;
+            case LONG:
+                return BIGINT;
+            case FLOAT:
+                return REAL;
+            case DOUBLE:
+                return DOUBLE;
+            case STRING:
+                return VARCHAR;
+            case BYTES:
+                return VARBINARY;
+            case INT_ARRAY:
+                return new ArrayType(INTEGER);
+            case LONG_ARRAY:
+                return new ArrayType(BIGINT);
+            case DOUBLE_ARRAY:
+                return new ArrayType(DOUBLE);
+            case STRING_ARRAY:
+                return new ArrayType(VARCHAR);
+            default:
+                break;
```

Comment:
Maybe fall-through instead?

Reply:
I just removed the default block, left conversation unresolved so you can take a look.

Reply:
Got an errorprone warning, will try a few other things out and update shortly.

Reply:
ErrorProne might complain if you don't have a case for each of the possible enumerations.
If we already cover all values of the enum then removing `default` and letting fall-through would've worked but it seems we're missing a case for `OBJECT`.

Let's keep as is.